### PR TITLE
Search functionality

### PR DESCRIPTION
**Pull request by ajohnson052**
Search results are now being populated with jQuery!

This also includes a method for adding and removing cases to and from the call list, but we will need to rework that part. 1) In the routes I originally included a user id in case we needed the option for a user (e.g. a manager) creating call lists for other users. It sounds like this isn't necessary. 2) Right now multiple users can add the same case to their call lists. 3) It needs some AJAX action.

### CODE REVIEW

**Comment by colinxfleming** [comment]
Heroic! Thanks @ajohnson052 ! I'll get this reviewed tomorrow. 

For 1), you're absolutely correct that we won't need to worry about this. CMs can and do flag particular cases for each other, and the Urgent functionality takes care of it the rest of the way, since those all show up at the bottom. 

For 2) This is dead on as well. One of the major problems we're trying to solve is case managers not being able to look at a case at the same time, so we definitely want users to be able to futz around with the same case simultaneously. 

It looks like the CI check is coming back red -- we'll want to resolve that before merging.

Fabulous work! Thanks for your hustle on this.

**Comment by ajohnson052** [comment]
My mistake on the testing! I forgot to run all of the new tests once I merged in master. I just had to sign in a user for the CasesControllerTest, and things appear to be running smoothly now. Looking forward to your comments!

**Comment by colinxfleming** [comment]
CI is coming back clean now! Thanks for fixing!

Taking a look at this now.

File: app/controllers/cases_controller.rb [hunk 1]
```diff
@@ -4,4 +4,20 @@ def index
     @p_case = @patient.cases.build
     @urgent_cases = Case.where(urgent_flag: true)
   end
+
+  def search
+    name_match = Patient.where(name: params[:search])
+    primary_match = Patient.where(primary_phone: params[:search])
+    secondary_match = Patient.where(secondary_phone: params[:search])
+    patients = name_match | primary_match | secondary_match
```
Comment (colinxfleming) [inline app/controllers/cases_controller.rb]: Had no idea that `|` was a concatenation operator for arrays! Learn something new every day. Thanks for showing me this!

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class UsersController < ApplicationController
+
+  def add_case
```
Comment (colinxfleming) [inline app/controllers/users_controller.rb]: This corresponds to creating call lists for others, right? Can we remove?

File: app/models/case.rb [hunk 3]
```diff
@@ -30,6 +30,7 @@ class Case
 	field :urgent_flag, type: Boolean
 
 	belongs_to :patient
+	belongs_to :user
```
Comment (colinxfleming) [inline app/models/case.rb]: probably strip this relationship as well?

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (colinxfleming) [inline app/controllers/users_controller.rb]: (Let me know if I'm wrong and we're keeping track of call list for an individual user with this method! Or if we can adapt it to keep call list in session)

File: app/views/cases/_search_results.html.erb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<h4>Search Results</h4>
```
Comment (colinxfleming) [inline app/views/cases/_search_results.html.erb]: :100: for making this a partial off the bat!

File: app/views/cases/index.html.erb [hunk 5]
```diff
@@ -41,28 +37,9 @@
...
-          <td><span class="glyphicon glyphicon-chevron-right" aria-hidden="true"></span></td>
-        </tr>
-        <tr>
-          <td>202-345-6677</td>
-          <td>Patty W</td>
-          <td>9 weeks</td>
-          <td>(sample case notes. something or other. detail)</td>
-          <td><span class="glyphicon glyphicon-chevron-right" aria-hidden="true"></span></td>
-        </tr>
+        <%= render partial: 'table_content', locals: {cases: current_user.cases} %>
```
Comment (colinxfleming) [inline app/views/cases/index.html.erb]: Ahhhh I see what's up here. Sorry for commenting before understanding! If this is tied to `current_user` it's solid.

File: config/routes.rb [hunk 6]
```diff
@@ -10,4 +10,7 @@
   end
   root :to => redirect('/users/sign_in')
   devise_for :users
+  patch 'users/:user_id/add_case/:id', to: 'users#add_case'
```
Comment (colinxfleming) [inline config/routes.rb]: Can you move these additional routes into the `authenticate :user do` block? Want to signify that they require being authenticated to do any of them.

**Comment by colinxfleming** [comment]
@ajohnson052 this is really great work. Nicely done! This is further along than where I thought it was -- really impressed that you got this out in only two hours. 

I think the only real change is moving the new routes into the `authenticate` block. Other than that, it looks good, works on my machine, and passes tests.

Fixes #103 
Fixes #102 
Fixes #109 

Partially implements #106 
Partially implements #108

**Comment by ajohnson052** [comment]
@colinxfleming I think addresses the authenticate user problem for add/remove calls. Let me know if there's anything I missed!

**Comment by colinxfleming** [comment]
@ajohnson052  thank you so much for your hustle on this (again)! A+ work.

**Pull request merged**